### PR TITLE
Poset refactoring

### PR DESCRIPTION
**Pull request by plt-amy**
Just rewriting `Algebra.Lattice`, `Algebra.Semilattice`, `Algebra.Frame`, `Cat.Thin.*`, and getting rid of `Relation.Order`. I'm not sure if the names `Cat.Order.*` are final: I didn't want to have to go and patch everything that depends on `Cat.Thin` yet (there are a few modules scattered throughout, like 7 or so) in one go so I put the new stuff in a different prefix for now.

### CODE REVIEW

File: src/Cat/Order/Base.lagda.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+module Poset {ℓ ℓ′} (P : Poset ℓ ℓ′) where
+  open Poset-on (P .snd) public
+
+  Ob : Type ℓ
+  Ob = ⌞ P ⌟
+
+  private variable
+    w x y z : ⌞ P ⌟
+
+  _≤⟨_⟩_ : (w : ⌞ P ⌟) → w ≤ x → x ≤ y → w ≤ y
```
Comment (TOTBWF) [inline src/Cat/Order/Base.lagda.md:162]: Should this live in the `Poset` record, or a `Poset.Reasoning` module?

File: src/Cat/Order/Cat.lagda.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+module actually formalises that connection by constructing a fully
+faithful functor from the category of posets into the category of [strict
+categories]. The construction of a category from a poset is entirely
+unsurprising, but it is lengthy, thus ending up in this module.
+
+[poset]: Cat.Order.Base.html
+[strict categories]: Cat.Instances.StrictCat.html
+
+```agda
+to-category : ∀ {ℓ ℓ′} → Posets.Ob → Precategory ℓ ℓ′
```
Comment (TOTBWF) [inline src/Cat/Order/Cat.lagda.md:23]: Maybe call this `poset->category`?

**Review by TOTBWF — APPROVED** [review]

File: src/Cat/Order/Base.lagda.md (hunk 1 above)
Comment (plt-amy) [reply src/Cat/Order/Base.lagda.md:162]: Hmmm good point, probably worth it to punt the similar stuff for Semilattices into a different module too

File: src/Cat/Order/Cat.lagda.md (hunk 2 above)
Comment (plt-amy) [reply src/Cat/Order/Cat.lagda.md:23]: Will do

**Pull request merged**